src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -183,6 +183,7 @@ public override Binder VisitMethodDeclaration(MethodDeclarationSyntax methodDecl
                     {
                         method = method ?? GetMethodSymbol(methodDecl, resultBinder);
                         isIteratorBody = method.IsIterator;
+                        resultBinder = WithExtensionReceiverParameterBinderIfNecessary(resultBinder, method);
                         resultBinder = new InMethodBinder(method, resultBinder);
```

Comment:
Wouldn't it be better to update `InMethodBinder` to handle extension receiver?

Reply:
> Wouldn't it be better to update InMethodBinder to handle extension receiver?

It depends. My goal in this PR is to quickly enable test scenarios for implementation methods. I'll add a prototype comment.